tsdb/head.go

```diff
@@ -53,13 +53,23 @@ var (
 	// ErrAppenderClosed is returned if an appender has already be successfully
 	// rolled back or committed.
 	ErrAppenderClosed = errors.New("appender closed")
+
+	WALReplayStatus walReplayStatus
```

Comment:
Is this global variable? I think that's might be not the best option. First of all, it will block people for running multiple Heads (e.g for test or any other purposes).

In fact we do this all the time in Thanos and Cortex projects.

Reply:
Yeah, in my working version of the codebase I had moved this into a field in the `Head` struct, added functions to modify the values that were concurrency safe, and also added the `WalReplayStats` (indirectly) to the `API` struct. I was just waiting on getting guidance on the head configuration thing so I could complete the work. Probably should have pushed as is - sorry for the trouble.

Reply:
ah, perfect, looing now!